plugin.rb

```diff
@@ -187,18 +191,20 @@ class ::Post
       def discourse_automation_topic_required_words
         return if self.post_type == Post.types[:small_action]
 
-        if topic.custom_fields['discourse_automation_id'].present?
-          automation = DiscourseAutomation::Automation.find(topic.custom_fields['discourse_automation_id'])
-          if automation&.script == DiscourseAutomation::Scriptable::TOPIC_REQUIRED_WORDS
-            words = automation.fields.find_by(name: 'words')
+        if topic.custom_fields[DiscourseAutomation::CUSTOM_FIELD].present?
+          topic.custom_fields[DiscourseAutomation::CUSTOM_FIELD].each do |automation_id|
+            automation = DiscourseAutomation::Automation.find_by(id: automation_id)
+            if automation&.script == DiscourseAutomation::Scriptable::TOPIC_REQUIRED_WORDS
+              words = automation.fields.find_by(name: 'words')
 
-            return if !words
+              return if !words
```

Comment:
@jjaffeux When you wrapped this code in a `#each` loop, did you mean this to be still a `return` or should it be changed to `next`?

Reply:
Indeed you are right, thanks

https://github.com/discourse/discourse-automation/pull/84